sources/packages/backend/apps/api/src/services/cas-invoice-batch/cas-invoice-batch-report.service.ts

```diff
@@ -0,0 +1,191 @@
+  async getCASInvoiceBatchesReport(
+    invoiceBatchId: number,
+  ): Promise<CASInvoiceBatchReport> {
+    const reportData = await this.getCASInvoiceBatchesReportData(
+      invoiceBatchId,
+    );
+    if (!reportData) {
+      throw new CustomNamedError(
+        `CAS invoice batch with ID ${invoiceBatchId} not found.`,
+        CAS_INVOICE_BATCH_NOT_FOUND,
+      );
+    }
+    const report = reportData.casInvoices.flatMap((invoice) => {
+      // Rows for a single invoice.
+      // One row per award will be generated.
+      const invoiceReportRows: Record<string, string>[] = [];
+      // Invoice master details. Each award will have this information.
+      const reportInvoiceMaster = this.createReportInvoiceMaster(invoice);
+      // Get all unique awards for the invoice. Each award will be a row in the report.
+      const awardCodes = new Set(
+        invoice.casInvoiceDetails.map(
+          (detail) => detail.casDistributionAccount.awardValueCode,
+        ),
+      );
+      // Get the awards distribution details to be appended to the same row.
+      awardCodes.forEach((awardCode) => {
+        const awardAccounts = invoice.casInvoiceDetails.filter(
+          (detail) =>
+            detail.casDistributionAccount.awardValueCode === awardCode,
+        );
+        const reportInvoiceRow = {
+          ...reportInvoiceMaster,
+          "Award Type": awardCode,
+        };
+        awardAccounts.forEach((account) => {
+          // Append the values for each operation code.
+          const operation = account.casDistributionAccount.operationCode;
+          reportInvoiceRow[`${operation} Amount`] =
+            account.valueAmount.toString();
+          reportInvoiceRow[`${operation} Account`] =
+            account.casDistributionAccount.distributionAccount;
+        });
+        invoiceReportRows.push(reportInvoiceRow);
+      });
+      return invoiceReportRows;
+    });
+    const reportCSVContent = unparse(report);
+    return {
+      batchName: reportData.batchName,
+      batchDate: reportData.batchDate,
+      reportCSVContent,
+    };
+  }
+
+  /**
+   * Creates a row containing the master details for an invoice.
+   * These details are common to all awards on the invoice.
+   * @param invoice the CAS invoice to generate the report row for.
+   * @returns a plain object with the report row data.
+   */
+  private createReportInvoiceMaster(
+    invoice: CASInvoice,
+  ): Record<string, string> {
+    // Student data for easy access.
+    const student =
+      invoice.disbursementReceipt.disbursementSchedule.studentAssessment
+        .application.student;
+    // Schedule data for easy access.
+    const schedule = invoice.disbursementReceipt.disbursementSchedule;
+    return {
+      "Invoice Number": invoice.invoiceNumber,
+      "Given Names": student.user.firstName,
+      "Last Name": student.user.lastName,
+      SIN: student.sinValidation.sin,
+      "Supplier Number": invoice.casSupplier.supplierNumber,
```

Comment:
Is the Site number not required?, I saw that in one place in the ticket, but not in other place. 
![image](https://github.com/user-attachments/assets/e5e43913-f98b-45c4-8dc7-f018f1e098c2)

Reply:
This image is from the analysis comments. Please rely on the business AC as a source of truth for this type of information. I have also updated the technical ACs earlier.